### ISSUE TITLE
Sign in to sync toolbar menu item, indicates I'm logout when I'm not

### ISSUE THREAD

**Issue by Amejia481** [mozilla-mobile/fenix#26072]
### Steps to reproduce

Prerequisites authenticate with your Firefox account. 

1. Kill Nightly.
2. Open Nightly.
3. Tab the 3 dotted menu. 

### Expected behaviour

The menu item should indicate that I'm authenticated 

### Actual behaviour

The menu item indicates I'm not authenticated 

### Android version

Android 12

### Firefox release type

Firefox Nightly

### Firefox version

104.0a1

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-21040)

**Comment by delia-pop** [comment]
I can reproduce the issue on the 07/18 Nightly with Google Pixel 6 (Android 12) and Huawei MediaPad M2 (Android 5.1.1).
After signing in, the "Sign in to sync" menu section is not updated with the account information. The section updates only after accessing it. 

https://user-images.githubusercontent.com/89388888/179533316-b71daec1-44df-496e-9f1a-a704a5f2d9c3.mp4

**Comment by Alexandru2909** [comment]
This seems to be a duplicate of https://github.com/mozilla-mobile/fenix/issues/25486. Also, https://github.com/mozilla-mobile/fenix/issues/26050 contains the details to fixing this behavior. @delia-pop can you confirm this?

**Comment by LaurentiuApahideanSV** [comment]
I tested the issue on Firefox Nightly 106.0a1 2022-08-26 and the issue no longer occurs. The "Sign in to sync" button is not shown until a sync is performed.

https://user-images.githubusercontent.com/50983519/186940313-d8352707-0a18-4d01-8b90-0238bc95e713.mp4

### Devices used:
- Google Pixel 6 (Android 13)
- Samsung Galaxy S22 Ultra (Android 12)

**Comment by Juraj-Masiar** [comment]
I'm not sure this is a proper fix.
You have a bug with user being in a "Not logged in" state when Firefox opens and you fix it by hiding the "Sign in to sync" button?

When I open Firefox, I'm not logged in.
I open any page, I'm still not logged in.
Then I open Settings (where I'm already logged in).
Then return back and suddenly I'm logged in.
![bug](https://user-images.githubusercontent.com/6358971/188107676-204b2299-7164-4e4c-8871-e6f768790bf0.png)

**Also, why should login info be hidden on the home screen and displayed only after visiting a page? That makes no sense.**

**Comment by mhammond** [comment]
I agree that fix is not ideal, but in my experience, I was actually signed in - it was just the toolbar state wrong, but "Settings" should have shown you signed in?

> Also, why should login info be hidden on the home screen

If this doesn't show when you *really* aren't signed in to sync then that's very bad.

**Comment by Juraj-Masiar** [comment]
But you see the similar behavior also in the video above in https://github.com/mozilla-mobile/fenix/issues/26072#issuecomment-1228640804, there is no `email@domain.xx` in the main menu, it appears only after he visits the page. And even that is probably "time-related".

I'm signed it all the time, I never sign out. Yet the state reported by Firefox UI is totally inconsistent. Why can't it read the last login state from storage and then update it after the sync.
The current implementation feels like a huge race-condition.

**Comment by mhammond** [comment]
Right - I agree 100% it's confusing, but I don't think there's a race condition - that main menu is plain wrong, but only that. We can and should fix it.

I really hope that main menu works ok for people who aren't actually signed in - they will not get a "first sync", so it might not appear!

**Comment by Juraj-Masiar** [comment]
Great!
I've been experimenting with this now, trying to recreate the state from https://github.com/mozilla-mobile/fenix/issues/26072#issuecomment-1235274656, where after visiting the page, the menu doesn't show my login email, even though I'm still signed in.
And I couldn't reproduce it... even after killing Nightly. Which was strange because this is happening to me ALL the time.

Then my Firefox updated (like it does every day) and then it reproduced just fine. No login info in any menu in any page, ONLY after I've visited Settings - where I didn't clicked anything, just opening it fixes the login state.
I think there is some special "cleanup" after each update that breaks this.
This may not be a "big deal" for normal users that updates once a month, but for Nightly users this is every day bug :(
